ldap/servers/plugins/sync/sync_util.c

```diff
@@ -487,7 +487,7 @@ sync_cookie_get_change_number(int lastnr, const char *uniqueid)
     LDAPControl **ctrls = NULL;
     
     ctrls = (LDAPControl **)slapi_ch_calloc(2, sizeof(LDAPControl *));
-    char *filter = slapi_ch_smprintf("(&(changenumber>=%d)(targetuniqueid=%s))", lastnr, uniqueid);
+    char *filter = slapi_ch_smprintf("(&(changenumber>=%d)(targetuniqueid=%s)(!(changenumber=%d)))", lastnr, uniqueid, lastnr);
```

Comment:
Instead of adding a not condition, why not use lastnr + 1?

Reply:
Initially I was thinking doing this. In fact one of the patch you did also modified a filter with nr+1.
It works but it makes some assumption on the changenumber (numeric and increased by one). I have no strong opinion on one option or the other. 

One consideration is that we realized that using retroCL to implement sync_repl had some drawbacks and that possibly in the future sync_repl will rely on the replication changelog. In such case, update ID will no longer be numeric/incremented by one.

But so many things will need to be changed in sync_repl if we move from retroCL to replication changelog that I agree with you that it is better to follow your proposal. I will update the patch.

Reply:
Yeah I think it would be a lot of work to move sync repl to the main changelog. It would effectively be a complete rewrite I think. So then the decision we make here won't affect a future change like that. Thanks for updating this!